packages/openai-compatible/src/map-openai-compatible-completion-logprobs.ts

```diff
@@ -0,0 +1,24 @@
+import { LanguageModelV1LogProbs } from '@ai-sdk/provider';
+
+type OpenAICompatibleCompletionLogProps = {
+  tokens: string[];
+  token_logprobs: number[];
+  top_logprobs: Record<string, number>[] | null;
+};
+
+export function mapOpenAICompatibleCompletionLogProbs(
+  logprobs: OpenAICompatibleCompletionLogProps | null | undefined,
+): LanguageModelV1LogProbs | undefined {
+  return logprobs?.tokens.map((token, index) => ({
+    token,
+    logprob: logprobs.token_logprobs[index],
+    topLogprobs: logprobs.top_logprobs
+      ? Object.entries(logprobs.top_logprobs[index]).map(
+          ([token, logprob]) => ({
+            token,
+            logprob,
+          }),
+        )
+      : [],
+  }));
+}
```

Comment:
Let's remove logprobs from the compatible provider. I'm leaning towards moving it into a provider extension since it's rarely supported.